src/DynamoCoreWpf/ViewModels/Search/SelectionNavigator.cs

```diff
@@ -107,6 +131,14 @@ internal void MoveSelection(NavigationDirection direction)
                             var group = category.MemberGroups.ElementAt(selectedMemberGroupIndex);
                             selectedMemberIndex = group.Members.Count() - 1;
                         }
+                        else // No place to move back. Clear selection. Select top result.
+                        {
+                            selection = GetSelectionFromIndices();
+                            selection.IsSelected = false;
+                            selection = null;
+                            topResult.IsSelected = true;
+                            return;
+                        }
```

Comment:
I'd suggest the following changes to avoid having condition where `selection == null`, in another word, `selection` should never be `null`. It can be the same as `topResult` when the selection is top result. In this case, the following data members should all be `-1`, like so:

```csharp
selectedCategoryIndex = -1;
selectedMemberGroupIndex = -1;
selectedMemberIndex = -1;
```

Then in `GetSelectionFromIndices` method, this is what you can do:

```csharp
private NodeSearchElementViewModel GetSelectionFromIndices()
{
    if ((selectedCategoryIndex == -1) && 
        (selectedMemberGroupIndex == -1) && 
        (selectedMemberIndex == -1))
    {
        // No selection, return topResult instead.
        return topResult;
    }

    // Bla bla...
}
```

Then in `MoveSelection`, this is what you can do:

```csharp
internal void MoveSelection(NavigationDirection direction)
{
    if (selection == topResult)
    {
        // We can only move forward...
        if (direction == NavigationDirection.Forward)
        {
            selectedCategoryIndex = 0;
            selectedMemberGroupIndex = 0;
            selectedMemberIndex = 0;

            selection.IsSelected = false;
            selection = GetSelectionFromIndices();
            selection.IsSelected = true;
         }

         return selection; // Otherwise, nothing changes.
    }
}
```

Does that make sense?